### PR TITLE
node pools: implement HTTP API to list jobs in pool

### PR DESCRIPTION
**Pull request by tgross**
Implements the HTTP API associated with the `NodePool.ListJobs` RPC, including the `api` package for the public API. Also fixes a bug in the RPC handler where we were missing handling of the special `all` node pool.

### CODE REVIEW

File: api/node_pools.go [hunk 1]
```diff
@@ -103,3 +104,15 @@ type NodePool struct {
...
+func (n *NodePools) ListJobs(poolName string, q *QueryOptions) ([]*JobListStub, *QueryMeta, error) {
+	var resp []*JobListStub
+	qm, err := n.client.query(
+		fmt.Sprintf("/v1/node/pool/%s/jobs", url.PathEscape(poolName)),
+		&resp, q)
+	if err != nil {
+		return nil, nil, err
+	}
+	return resp, qm, nil
+}
```
Comment (lgfa29) [inline api/node_pools.go:118]: It would be nice to keep this up there along with the other methods, or maybe move the structs up?

**Review by lgfa29 — APPROVED** [review]
LGTM.

We need to document this endpoint, which could be done here or is a separate PR.

File: command/agent/node_pool_endpoint.go [hunk 2]
```diff
@@ -125,6 +128,10 @@ func (s *HTTPServer) nodePoolDelete(resp http.ResponseWriter, req *http.Request,
 }
 
 func (s *HTTPServer) nodePoolNodesList(resp http.ResponseWriter, req *http.Request, poolName string) (interface{}, error) {
+	if req.Method != http.MethodGet {
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
```
Comment (tgross) [inline command/agent/node_pool_endpoint.go:133]: @lgfa29 when I rebased on `main` I added this assertion so we don't list on `POST` or whatever.

Comment (lgfa29) [reply command/agent/node_pool_endpoint.go:133]: Ah good catch, thanks!

**Pull request merged**